src/Sylius/Bundle/ChannelBundle/Behat/ChannelContext.php

```diff
@@ -144,15 +145,35 @@ public function theDefaultChannelHasFollowingConfiguration(TableNode $table)
     }
 
     /**
+     * @Given /^there is a disabled channel "([^""]*)"$/
+     */
+    public function thereIsDisabledChannel($name)
+    {
+        $this->thereIsChannel($name, $name, null, false);
+    }
+
+    /**
+     * @Given /^there is an enabled channel "([^""]*)"$/
+     */
+    public function thereIsEnabledChannel($name)
+    {
+        $this->thereIsChannel($name, $name, null, true);
+    }
+
+    /**
      * @Given /^There is channel "([^""]*)" named "([^""]*)" for url "([^""]*)"$/
      */
-    public function thereIsChannel($code, $name, $url, $locales = null, $currencies = "EUR", $shippingMethods = null, $paymentMethos = null, $flush = true)
+    public function thereIsChannel($code, $name, $url, $enabled = true, $locales = null, $currencies = "EUR", $shippingMethods = null, $paymentMethos = null, $flush = true)
     {
         /* @var $channel ChannelInterface */
-        $channel = $this->getRepository('channel')->createNew();
-        $channel->setCode($code);
-        $channel->setName($name);
-        $channel->setUrl($url);
+        if (null === $channel = $this->getRepository('channel')->findOneBy(array('code' => $code))) {
```

Comment:
Using strict comparsion just for sake of using is wrong anyway, if that method will return `false`, `0`, empty string/array, that code will die with fatal error instead of correctly creating new channel...

@michalmarcinkowski Going further with your thinking, what's more redeable:
```php
if (!$value) {}
// vs
if (null === $value || 0 === $value || false === $value || '' === $value/* etc. */) {}
```
:stuck_out_tongue_winking_eye: